SilverSolver/assignment1/Q2.py

```diff
@@ -0,0 +1,44 @@
+class single_linked_node:
+
+    def __init__(self, value=None, child=None):
+        self.child = child
+        self.value = value
+
+    def __getitem__(self, k):
+        k = -k
+        list_len = 0
+        current_node = self
+        while (current_node):
+            list_len += 1
+            current_node = current_node.child
+        element_num = list_len - k
+        if element_num < 0 or element_num > list_len:
+            raise IndexError('wrong k')
```

Comment:
Think about the poor developer that needs to make the software work for tomorrow and software breaks all the time, and the only way they can debug is via the logs, and the logs says 'wrong k' :)
Here the problem is that the error message does not define 'wrong'. IMHO, here is a better message for them:

raise IndexError("Index '%s' is bigger than the length of the list '%s'" % (k, list_len))